modin/engines/ray/utils.py

```diff
@@ -25,3 +29,100 @@ def handle_ray_task_error(e):
                 else:
                     raise att_err
     raise e
+
+
+# Register a fix import function to run on all_workers including the driver.
+# This is a hack solution to fix #647, #746
+def _move_stdlib_ahead_of_site_packages(*args):
+    site_packages_path = None
+    site_packages_path_index = -1
+    for i, path in enumerate(sys.path):
+        if sys.exec_prefix in path and path.endswith("site-packages"):
+            site_packages_path = path
+            site_packages_path_index = i
+            # break on first found
+            break
+
+    if site_packages_path is not None:
+        # stdlib packages layout as follows:
+        # - python3.x
+        #   - typing.py
+        #   - site-packages/
+        #     - pandas
+        # So extracting the dirname of the site_packages can point us
+        # to the directory containing standard libraries.
+        sys.path.insert(site_packages_path_index, os.path.dirname(site_packages_path))
+
+
+# Register a fix to import pandas on all workers before running tasks.
+# This prevents a race condition between two threads deserializing functions
+# and trying to import pandas at the same time.
+def _import_pandas(*args):
+    import pandas  # noqa F401
+
+
+def initialize_ray(cluster=None, redis_address=None, redis_password=None):
+    import ray
+
+    """Initializes ray based on environment variables and internal defaults."""
+    if threading.current_thread().name == "MainThread":
+        import secrets
+
+        plasma_directory = None
+        num_cpus = os.environ.get("MODIN_CPUS", None) or multiprocessing.cpu_count()
+        cluster = cluster or os.environ.get("MODIN_RAY_CLUSTER", None)
+        redis_address = redis_address or os.environ.get("MODIN_REDIS_ADDRESS", None)
+        redis_password = redis_password or secrets.token_hex(16)
+
+        if cluster == "True" and redis_address is not None:
+            # We only start ray in a cluster setting for the head node.
+            ray.init(
+                include_webui=False,
+                ignore_reinit_error=True,
+                address=redis_address,
+                redis_password=redis_password,
+                logging_level=100,
+            )
+        elif cluster is None:
```

Comment:
what if `cluster` is neither `"True"` nor `None`?.. there's no `else:` branch there, so `ray.init()` won't be run in that case which seems strange - there should be at least a warning (maybe an error) IMHO

Reply:
Yes, there should be and error. Good catch!

Reply:
`raise ValueError` used in the case now.